app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  private
+
+  def require_admin
+    return if current_user&.admin?
+
+    redirect_to root_path, alert: t('admin.errors.not_authorized')
+  end
```

Comment:
unlessは読みづらくなるからあまり使わない方が良いんだけど、お好みで以下のように書けます。
ただ、自分としては元のままでいいかなと思います〜

```suggestion
  def require_admin
    redirect_to root_path, alert: t('admin.errors.not_authorized') unless current_user&.admin?
  end
```